frog/server/stream.js

```diff
@@ -1,40 +1,42 @@
 // @flow
 
 import { Meteor } from 'meteor/meteor';
-import { generateReactiveFn, splitPathObject } from 'frog-utils';
+import { generateReactiveFn } from 'frog-utils';
 
+import { Activities } from '/imports/api/activities';
+import { activityTypesObj } from '/imports/activityTypes';
 import { serverConnection } from './share-db-manager';
 import { SharedbCache } from '../imports/api/cache';
 import LearningItem from '../imports/ui/LearningItem';
 
-const safelyInsertObject = (doc, dataFn, path, value, instanceId) => {
-  const { insertObject, insertPath } = splitPathObject(doc.data, path, value);
-  dataFn.objInsert(insertObject, insertPath);
-  dataFn.objInsert(instanceId, [
-    ...path.slice(0, path.length - 1),
-    'instanceId'
-  ]);
-};
-
 Meteor.methods({
-  stream: (activity, instanceId, path, value) => {
+  stream: (activity, instanceId, value) => {
     if (activity.streamTarget) {
+      const target = Activities.findOne(activity.streamTarget);
       const docId = activity.streamTarget + '/all';
+      const mergeFunction = activityTypesObj[target.activityType].mergeFunction;
+      if (!mergeFunction) {
+        return;
+      }
+      const toSend = {
+        data: { '1': { ...value, id: '1' } },
```

Comment:
For example the merge Function of ac-chat expects an array here, but I think we did agree on using keyed objects instead of array. If that's still what we want, let's look at why ac-chat merge works on array and let's update as much as we can to make activities stream compatible. 

Why is the id here `1` instead of the id of the original LI ?